<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
      <link rel="shortcut icon" href="../img/favicon.ico" />
    <title>Get variants - pandasGWAS Documentation</title>
    <link rel="stylesheet" href="../css/theme.css" />
    <link rel="stylesheet" href="../css/theme_extra.css" />
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/styles/github.min.css" />
        <link href="../assets/_mkdocstrings.css" rel="stylesheet" />
    
      <script>
        // Current page data
        var mkdocs_page_name = "Get variants";
        var mkdocs_page_input_path = "get_variants.md";
        var mkdocs_page_url = null;
      </script>
    
    <!--[if lt IE 9]>
      <script src="../js/html5shiv.min.js"></script>
    <![endif]-->
      <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/highlight.min.js"></script>
      <script>hljs.highlightAll();</script> 
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
    <div class="wy-side-scroll">
      <div class="wy-side-nav-search">
          <a href=".." class="icon icon-home"> pandasGWAS Documentation
        </a><div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
      <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="..">pandasGWAS Documentation</a>
                </li>
              </ul>
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../Association/">Association</a>
                </li>
              </ul>
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../Browser/">Browser</a>
                </li>
              </ul>
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../Study/">Study</a>
                </li>
              </ul>
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../Trait/">Trait</a>
                </li>
              </ul>
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../Variant/">Variant</a>
                </li>
              </ul>
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../get_associations/">Get associations</a>
                </li>
              </ul>
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../get_studies/">Get studies</a>
                </li>
              </ul>
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../get_traits/">Get traits</a>
                </li>
              </ul>
              <ul class="current">
                <li class="toctree-l1 current"><a class="reference internal current" href="./">Get variants</a>
    <ul class="current">
    <li class="toctree-l2"><a class="reference internal" href="#pandasgwas.get_variants.get_variants">get_variants</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#pandasgwas.get_variants.get_variants_all">get_variants_all</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#pandasgwas.get_variants.get_variants_by_association_id">get_variants_by_association_id</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#pandasgwas.get_variants.get_variants_by_cytogenetic_band">get_variants_by_cytogenetic_band</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#pandasgwas.get_variants.get_variants_by_efo_id">get_variants_by_efo_id</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#pandasgwas.get_variants.get_variants_by_efo_trait">get_variants_by_efo_trait</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#pandasgwas.get_variants.get_variants_by_gene_name">get_variants_by_gene_name</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#pandasgwas.get_variants.get_variants_by_genomic_range">get_variants_by_genomic_range</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#pandasgwas.get_variants.get_variants_by_pubmed_id">get_variants_by_pubmed_id</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#pandasgwas.get_variants.get_variants_by_reported_trait">get_variants_by_reported_trait</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#pandasgwas.get_variants.get_variants_by_study_id">get_variants_by_study_id</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#pandasgwas.get_variants.get_variants_by_variant_id">get_variants_by_variant_id</a>
    </li>
    </ul>
                </li>
              </ul>
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../set_operation/">Set operation</a>
                </li>
              </ul>
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../summary_statistics/">Summary statistics</a>
                </li>
              </ul>
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../utility/">Utility</a>
                </li>
              </ul>
      </div>
    </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">
      <nav class="wy-nav-top" role="navigation" aria-label="Mobile navigation menu">
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="..">pandasGWAS Documentation</a>
        
      </nav>
      <div class="wy-nav-content">
        <div class="rst-content"><div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href=".." class="icon icon-home" aria-label="Docs"></a></li>
      <li class="breadcrumb-item active">Get variants</li>
    <li class="wy-breadcrumbs-aside">
    </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
            <div class="section" itemprop="articleBody">
              
                <div class="doc doc-object doc-module">



<h2 id="pandasgwas.get_variants" class="doc doc-heading">
          <span class="doc doc-object-name doc-module-name">pandasgwas.get_variants</span>


</h2>

  <div class="doc doc-contents first">

  

  <div class="doc doc-children">










<div class="doc doc-object doc-function">



<h3 id="pandasgwas.get_variants.get_variants" class="doc doc-heading">
          <span class="doc doc-object-name doc-function-name">get_variants</span>


</h3>
<div class="doc-signature highlight"><pre><span></span><code><span class="n">get_variants</span><span class="p">(</span>
    <span class="n">study_id</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
    <span class="n">association_id</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
    <span class="n">variant_id</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
    <span class="n">efo_id</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
    <span class="n">pubmed_id</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
    <span class="n">genomic_range</span><span class="p">:</span> <span class="n">List</span><span class="p">[</span><span class="nb">int</span><span class="p">]</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
    <span class="n">gene_name</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
    <span class="n">efo_trait</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
    <span class="n">reported_trait</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
    <span class="n">set_operation</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;bind&quot;</span><span class="p">,</span>
    <span class="n">interactive</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span>
<span class="p">)</span> <span class="o">-&gt;</span> <span class="n">Variant</span>
</code></pre></div>

  <div class="doc doc-contents ">
  
      <p>Retrieves Single Nucleotide Polymorphisms via the NHGRI-EBI GWAS Catalog REST API. The REST API is queried multiple times with the criteria passed as arguments. By default all Single Nucleotide Polymorphisms that match the criteria supplied in the arguments are retrieved: this corresponds to the default set_operation set to 'bind', If you rather have only the Single Nucleotide Polymorphisms that match simultaneously all criteria provided, then set set_operation to 'intersection'.</p>
<pre><code class="language-Python">from pandasgwas.get_variants import get_variants_by_variant_id
snps = get_variants(variant_id='rs7744020')
</code></pre>



<table class="field-list">
  <colgroup>
    <col class="field-name" />
    <col class="field-body" />
  </colgroup>
  <tbody valign="top">
    <tr class="field">
      <th class="field-name">Parameters:</th>
      <td class="field-body">
        <ul class="first simple">
            <li>
              <b><code>study_id</code></b>
                  (<code>str</code>, default:
                      <code>None</code>
)
              –
              <div class="doc-md-description">
                
              </div>
            </li>
            <li>
              <b><code>association_id</code></b>
                  (<code>str</code>, default:
                      <code>None</code>
)
              –
              <div class="doc-md-description">
                <p>Association identifier, associationId in Association</p>
              </div>
            </li>
            <li>
              <b><code>variant_id</code></b>
                  (<code>str</code>, default:
                      <code>None</code>
)
              –
              <div class="doc-md-description">
                <p>Single Nucleotide Polymorphism identifier, rsId in Single Nucleotide Polymorphism</p>
              </div>
            </li>
            <li>
              <b><code>efo_id</code></b>
                  (<code>str</code>, default:
                      <code>None</code>
)
              –
              <div class="doc-md-description">
                <p>EFO Trait identifier, shortForm in EFO Trait</p>
              </div>
            </li>
            <li>
              <b><code>pubmed_id</code></b>
                  (<code>str</code>, default:
                      <code>None</code>
)
              –
              <div class="doc-md-description">
                <p>PubMed identifier</p>
              </div>
            </li>
            <li>
              <b><code>genomic_range</code></b>
                  (<code><span title="typing.List">List</span>[int]</code>, default:
                      <code>None</code>
)
              –
              <div class="doc-md-description">
                <p>A List of three elements. These elements are: 'chromosome', 'start' and 'end' in function <a class="autorefs autorefs-internal" href="#pandasgwas.get_variants.get_variants_by_genomic_range">get_variants_by_genomic_range</a></p>
              </div>
            </li>
            <li>
              <b><code>gene_name</code></b>
                  (<code>str</code>, default:
                      <code>None</code>
)
              –
              <div class="doc-md-description">
                <p>Gene names</p>
              </div>
            </li>
            <li>
              <b><code>efo_trait</code></b>
                  (<code>str</code>, default:
                      <code>None</code>
)
              –
              <div class="doc-md-description">
                <p>Trait description</p>
              </div>
            </li>
            <li>
              <b><code>reported_trait</code></b>
                  (<code>str</code>, default:
                      <code>None</code>
)
              –
              <div class="doc-md-description">
                <p>Trait are reported by the original authors of the study</p>
              </div>
            </li>
            <li>
              <b><code>set_operation</code></b>
                  (<code>str</code>, default:
                      <code>&#39;bind&#39;</code>
)
              –
              <div class="doc-md-description">
                <p>"bind" or "intersection"</p>
              </div>
            </li>
            <li>
              <b><code>interactive</code></b>
                  (<code>bool</code>, default:
                      <code>True</code>
)
              –
              <div class="doc-md-description">
                <p>Indicates whether to run in interactive mode, when the value is True and the query data is divided into many pages, the function will prompt whether to continue the download.</p>
              </div>
            </li>
        </ul>
      </td>
    </tr>
  </tbody>
</table>


<table class="field-list">
  <colgroup>
    <col class="field-name" />
    <col class="field-body" />
  </colgroup>
  <tbody valign="top">
    <tr class="field">
    <th class="field-name">Returns:</th>
    <td class="field-body">
      <ul class="first simple">
          <li>
                <code><a class="autorefs autorefs-internal" title="pandasgwas.Variant.Variant" href="../Variant/#pandasgwas.Variant.Variant">Variant</a></code>
            –
            <div class="doc-md-description">
              <p>A Variant object</p>
            </div>
          </li>
      </ul>
    </td>
    </tr>
  </tbody>
</table>
  </div>

</div>


<div class="doc doc-object doc-function">



<h3 id="pandasgwas.get_variants.get_variants_all" class="doc doc-heading">
          <span class="doc doc-object-name doc-function-name">get_variants_all</span>


</h3>
<div class="doc-signature highlight"><pre><span></span><code><span class="n">get_variants_all</span><span class="p">(</span><span class="n">interactive</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">True</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">Variant</span>
</code></pre></div>

  <div class="doc doc-contents ">
  
      <p>Get all Single Nucleotide Polymorphisms</p>
<pre><code class="language-Python">from pandasgwas.get_variants import get_variants_all
snps = get_variants_all()
</code></pre>



<table class="field-list">
  <colgroup>
    <col class="field-name" />
    <col class="field-body" />
  </colgroup>
  <tbody valign="top">
    <tr class="field">
      <th class="field-name">Parameters:</th>
      <td class="field-body">
        <ul class="first simple">
            <li>
              <b><code>interactive</code></b>
                  (<code>bool</code>, default:
                      <code>True</code>
)
              –
              <div class="doc-md-description">
                <p>Indicates whether to run in interactive mode, when the value is True and the query data is divided into many pages, the function will prompt whether to continue the download.</p>
              </div>
            </li>
        </ul>
      </td>
    </tr>
  </tbody>
</table>


<table class="field-list">
  <colgroup>
    <col class="field-name" />
    <col class="field-body" />
  </colgroup>
  <tbody valign="top">
    <tr class="field">
    <th class="field-name">Returns:</th>
    <td class="field-body">
      <ul class="first simple">
          <li>
                <code><a class="autorefs autorefs-internal" title="pandasgwas.Variant.Variant" href="../Variant/#pandasgwas.Variant.Variant">Variant</a></code>
            –
            <div class="doc-md-description">
              <p>A Variant object</p>
            </div>
          </li>
      </ul>
    </td>
    </tr>
  </tbody>
</table>
  </div>

</div>


<div class="doc doc-object doc-function">



<h3 id="pandasgwas.get_variants.get_variants_by_association_id" class="doc doc-heading">
          <span class="doc doc-object-name doc-function-name">get_variants_by_association_id</span>


</h3>
<div class="doc-signature highlight"><pre><span></span><code><span class="n">get_variants_by_association_id</span><span class="p">(</span>
    <span class="n">association_id</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span> <span class="n">interactive</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">True</span>
<span class="p">)</span> <span class="o">-&gt;</span> <span class="n">Variant</span>
</code></pre></div>

  <div class="doc doc-contents ">
  
      <p>Get GWAS Catalog Single Nucleotide Polymorphisms by by their Association identifier</p>
<pre><code class="language-Python">from pandasgwas.get_variants import get_variants_by_association_id
snps = get_variants_by_association_id('16603')
</code></pre>



<table class="field-list">
  <colgroup>
    <col class="field-name" />
    <col class="field-body" />
  </colgroup>
  <tbody valign="top">
    <tr class="field">
      <th class="field-name">Parameters:</th>
      <td class="field-body">
        <ul class="first simple">
            <li>
              <b><code>association_id</code></b>
                  (<code>str</code>)
              –
              <div class="doc-md-description">
                <p>Association identifier, associationId in Association</p>
              </div>
            </li>
            <li>
              <b><code>interactive</code></b>
                  (<code>bool</code>, default:
                      <code>True</code>
)
              –
              <div class="doc-md-description">
                <p>Indicates whether to run in interactive mode, when the value is True and the query data is divided into many pages, the function will prompt whether to continue the download.</p>
              </div>
            </li>
        </ul>
      </td>
    </tr>
  </tbody>
</table>


<table class="field-list">
  <colgroup>
    <col class="field-name" />
    <col class="field-body" />
  </colgroup>
  <tbody valign="top">
    <tr class="field">
    <th class="field-name">Returns:</th>
    <td class="field-body">
      <ul class="first simple">
          <li>
                <code><a class="autorefs autorefs-internal" title="pandasgwas.Variant.Variant" href="../Variant/#pandasgwas.Variant.Variant">Variant</a></code>
            –
            <div class="doc-md-description">
              <p>A Variant object</p>
            </div>
          </li>
      </ul>
    </td>
    </tr>
  </tbody>
</table>
  </div>

</div>


<div class="doc doc-object doc-function">



<h3 id="pandasgwas.get_variants.get_variants_by_cytogenetic_band" class="doc doc-heading">
          <span class="doc doc-object-name doc-function-name">get_variants_by_cytogenetic_band</span>


</h3>
<div class="doc-signature highlight"><pre><span></span><code><span class="n">get_variants_by_cytogenetic_band</span><span class="p">(</span>
    <span class="n">cytogenetic_band</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span> <span class="n">interactive</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">True</span>
<span class="p">)</span> <span class="o">-&gt;</span> <span class="n">Variant</span>
</code></pre></div>

  <div class="doc doc-contents ">
  
      <p>Get GWAS Catalog Single Nucleotide Polymorphisms by cytogenetic band.</p>
<pre><code class="language-Python">from pandasgwas.get_variants import get_variants_by_cytogenetic_band
snps = get_variants_by_cytogenetic_band('1p36.32')
</code></pre>



<table class="field-list">
  <colgroup>
    <col class="field-name" />
    <col class="field-body" />
  </colgroup>
  <tbody valign="top">
    <tr class="field">
      <th class="field-name">Parameters:</th>
      <td class="field-body">
        <ul class="first simple">
            <li>
              <b><code>cytogenetic_band</code></b>
                  (<code>str</code>)
              –
              <div class="doc-md-description">
                <p>Cytogenetic bands of the form '1p36.11'</p>
              </div>
            </li>
            <li>
              <b><code>interactive</code></b>
                  (<code>bool</code>, default:
                      <code>True</code>
)
              –
              <div class="doc-md-description">
                <p>Indicates whether to run in interactive mode, when the value is True and the query data is divided into many pages, the function will prompt whether to continue the download.</p>
              </div>
            </li>
        </ul>
      </td>
    </tr>
  </tbody>
</table>


<table class="field-list">
  <colgroup>
    <col class="field-name" />
    <col class="field-body" />
  </colgroup>
  <tbody valign="top">
    <tr class="field">
    <th class="field-name">Returns:</th>
    <td class="field-body">
      <ul class="first simple">
          <li>
                <code><a class="autorefs autorefs-internal" title="pandasgwas.Variant.Variant" href="../Variant/#pandasgwas.Variant.Variant">Variant</a></code>
            –
            <div class="doc-md-description">
              <p>A Variant object</p>
            </div>
          </li>
      </ul>
    </td>
    </tr>
  </tbody>
</table>
  </div>

</div>


<div class="doc doc-object doc-function">



<h3 id="pandasgwas.get_variants.get_variants_by_efo_id" class="doc doc-heading">
          <span class="doc doc-object-name doc-function-name">get_variants_by_efo_id</span>


</h3>
<div class="doc-signature highlight"><pre><span></span><code><span class="n">get_variants_by_efo_id</span><span class="p">(</span>
    <span class="n">efo_id</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span> <span class="n">interactive</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">True</span>
<span class="p">)</span> <span class="o">-&gt;</span> <span class="n">Variant</span>
</code></pre></div>

  <div class="doc doc-contents ">
  
      <p>Get GWAS Catalog Single Nucleotide Polymorphisms by their EFO Trait identifier</p>
<pre><code class="language-Python">from pandasgwas.get_variants import get_variants_by_efo_id
snps = get_variants_by_efo_id('EFO_0001065')
</code></pre>



<table class="field-list">
  <colgroup>
    <col class="field-name" />
    <col class="field-body" />
  </colgroup>
  <tbody valign="top">
    <tr class="field">
      <th class="field-name">Parameters:</th>
      <td class="field-body">
        <ul class="first simple">
            <li>
              <b><code>efo_id</code></b>
                  (<code>str</code>)
              –
              <div class="doc-md-description">
                <p>EFO Trait identifier, shortForm in EFO Trait</p>
              </div>
            </li>
            <li>
              <b><code>interactive</code></b>
                  (<code>bool</code>, default:
                      <code>True</code>
)
              –
              <div class="doc-md-description">
                <p>Indicates whether to run in interactive mode, when the value is True and the query data is divided into many pages, the function will prompt whether to continue the download.</p>
              </div>
            </li>
        </ul>
      </td>
    </tr>
  </tbody>
</table>


<table class="field-list">
  <colgroup>
    <col class="field-name" />
    <col class="field-body" />
  </colgroup>
  <tbody valign="top">
    <tr class="field">
    <th class="field-name">Returns:</th>
    <td class="field-body">
      <ul class="first simple">
          <li>
                <code><a class="autorefs autorefs-internal" title="pandasgwas.Variant.Variant" href="../Variant/#pandasgwas.Variant.Variant">Variant</a></code>
            –
            <div class="doc-md-description">
              <p>A Variant object</p>
            </div>
          </li>
      </ul>
    </td>
    </tr>
  </tbody>
</table>
  </div>

</div>


<div class="doc doc-object doc-function">



<h3 id="pandasgwas.get_variants.get_variants_by_efo_trait" class="doc doc-heading">
          <span class="doc doc-object-name doc-function-name">get_variants_by_efo_trait</span>


</h3>
<div class="doc-signature highlight"><pre><span></span><code><span class="n">get_variants_by_efo_trait</span><span class="p">(</span>
    <span class="n">efo_trait</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span> <span class="n">interactive</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">True</span>
<span class="p">)</span> <span class="o">-&gt;</span> <span class="n">Variant</span>
</code></pre></div>

  <div class="doc doc-contents ">
  
      <p>Get GWAS Catalog Single Nucleotide Polymorphisms that match trait description</p>
<pre><code class="language-Python">from pandasgwas.get_variants import get_variants_by_efo_trait
snps = get_variants_by_efo_trait('MHPG measurement')
</code></pre>



<table class="field-list">
  <colgroup>
    <col class="field-name" />
    <col class="field-body" />
  </colgroup>
  <tbody valign="top">
    <tr class="field">
      <th class="field-name">Parameters:</th>
      <td class="field-body">
        <ul class="first simple">
            <li>
              <b><code>efo_trait</code></b>
                  (<code>str</code>)
              –
              <div class="doc-md-description">
                <p>Trait description</p>
              </div>
            </li>
            <li>
              <b><code>interactive</code></b>
                  (<code>bool</code>, default:
                      <code>True</code>
)
              –
              <div class="doc-md-description">
                <p>Indicates whether to run in interactive mode, when the value is True and the query data is divided into many pages, the function will prompt whether to continue the download.</p>
              </div>
            </li>
        </ul>
      </td>
    </tr>
  </tbody>
</table>


<table class="field-list">
  <colgroup>
    <col class="field-name" />
    <col class="field-body" />
  </colgroup>
  <tbody valign="top">
    <tr class="field">
    <th class="field-name">Returns:</th>
    <td class="field-body">
      <ul class="first simple">
          <li>
                <code><a class="autorefs autorefs-internal" title="pandasgwas.Variant.Variant" href="../Variant/#pandasgwas.Variant.Variant">Variant</a></code>
            –
            <div class="doc-md-description">
              <p>A Variant object</p>
            </div>
          </li>
      </ul>
    </td>
    </tr>
  </tbody>
</table>
  </div>

</div>


<div class="doc doc-object doc-function">



<h3 id="pandasgwas.get_variants.get_variants_by_gene_name" class="doc doc-heading">
          <span class="doc doc-object-name doc-function-name">get_variants_by_gene_name</span>


</h3>
<div class="doc-signature highlight"><pre><span></span><code><span class="n">get_variants_by_gene_name</span><span class="p">(</span>
    <span class="n">gene_name</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span> <span class="n">interactive</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">True</span>
<span class="p">)</span> <span class="o">-&gt;</span> <span class="n">Variant</span>
</code></pre></div>

  <div class="doc doc-contents ">
  
      <p>Get GWAS Catalog Single Nucleotide Polymorphisms gene name</p>
<pre><code class="language-Python">from pandasgwas.get_variants import get_variants_by_gene_name
snps = get_variants_by_gene_name('KIAA0319')
</code></pre>



<table class="field-list">
  <colgroup>
    <col class="field-name" />
    <col class="field-body" />
  </colgroup>
  <tbody valign="top">
    <tr class="field">
      <th class="field-name">Parameters:</th>
      <td class="field-body">
        <ul class="first simple">
            <li>
              <b><code>gene_name</code></b>
                  (<code>str</code>)
              –
              <div class="doc-md-description">
                <p>Gene names</p>
              </div>
            </li>
            <li>
              <b><code>interactive</code></b>
                  (<code>bool</code>, default:
                      <code>True</code>
)
              –
              <div class="doc-md-description">
                <p>Indicates whether to run in interactive mode, when the value is True and the query data is divided into many pages, the function will prompt whether to continue the download.</p>
              </div>
            </li>
        </ul>
      </td>
    </tr>
  </tbody>
</table>


<table class="field-list">
  <colgroup>
    <col class="field-name" />
    <col class="field-body" />
  </colgroup>
  <tbody valign="top">
    <tr class="field">
    <th class="field-name">Returns:</th>
    <td class="field-body">
      <ul class="first simple">
          <li>
                <code><a class="autorefs autorefs-internal" title="pandasgwas.Variant.Variant" href="../Variant/#pandasgwas.Variant.Variant">Variant</a></code>
            –
            <div class="doc-md-description">
              <p>A Variant object</p>
            </div>
          </li>
      </ul>
    </td>
    </tr>
  </tbody>
</table>
  </div>

</div>


<div class="doc doc-object doc-function">



<h3 id="pandasgwas.get_variants.get_variants_by_genomic_range" class="doc doc-heading">
          <span class="doc doc-object-name doc-function-name">get_variants_by_genomic_range</span>


</h3>
<div class="doc-signature highlight"><pre><span></span><code><span class="n">get_variants_by_genomic_range</span><span class="p">(</span>
    <span class="n">chromosome</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
    <span class="n">start</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span>
    <span class="n">end</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span>
    <span class="n">interactive</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span>
<span class="p">)</span> <span class="o">-&gt;</span> <span class="n">Variant</span>
</code></pre></div>

  <div class="doc doc-contents ">
  
      <p>Get GWAS Catalog Single Nucleotide Polymorphisms by genomic range</p>
<pre><code class="language-Python">from pandasgwas.get_variants import get_variants_by_genomic_range
snps = get_variants_by_genomic_range('1', 2300001, 5300000)
</code></pre>



<table class="field-list">
  <colgroup>
    <col class="field-name" />
    <col class="field-body" />
  </colgroup>
  <tbody valign="top">
    <tr class="field">
      <th class="field-name">Parameters:</th>
      <td class="field-body">
        <ul class="first simple">
            <li>
              <b><code>chromosome</code></b>
                  (<code>str</code>)
              –
              <div class="doc-md-description">
                <p>Human chromosome names: autosomal and sexual chromosomes only, i.e., 1--22, X and Y</p>
              </div>
            </li>
            <li>
              <b><code>start</code></b>
                  (<code>int</code>)
              –
              <div class="doc-md-description">
                <p>Start position of range (starts at 1).</p>
              </div>
            </li>
            <li>
              <b><code>end</code></b>
                  (<code>int</code>)
              –
              <div class="doc-md-description">
                <p>End position of range (inclusive).</p>
              </div>
            </li>
            <li>
              <b><code>interactive</code></b>
                  (<code>bool</code>, default:
                      <code>True</code>
)
              –
              <div class="doc-md-description">
                <p>Indicates whether to run in interactive mode, when the value is True and the query data is divided into many pages, the function will prompt whether to continue the download.</p>
              </div>
            </li>
        </ul>
      </td>
    </tr>
  </tbody>
</table>


<table class="field-list">
  <colgroup>
    <col class="field-name" />
    <col class="field-body" />
  </colgroup>
  <tbody valign="top">
    <tr class="field">
    <th class="field-name">Returns:</th>
    <td class="field-body">
      <ul class="first simple">
          <li>
                <code><a class="autorefs autorefs-internal" title="pandasgwas.Variant.Variant" href="../Variant/#pandasgwas.Variant.Variant">Variant</a></code>
            –
            <div class="doc-md-description">
              <p>A Variant object</p>
            </div>
          </li>
      </ul>
    </td>
    </tr>
  </tbody>
</table>
  </div>

</div>


<div class="doc doc-object doc-function">



<h3 id="pandasgwas.get_variants.get_variants_by_pubmed_id" class="doc doc-heading">
          <span class="doc doc-object-name doc-function-name">get_variants_by_pubmed_id</span>


</h3>
<div class="doc-signature highlight"><pre><span></span><code><span class="n">get_variants_by_pubmed_id</span><span class="p">(</span>
    <span class="n">pubmed_id</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span> <span class="n">interactive</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">True</span>
<span class="p">)</span> <span class="o">-&gt;</span> <span class="n">Variant</span>
</code></pre></div>

  <div class="doc doc-contents ">
  
      <p>Get GWAS Catalog Single Nucleotide Polymorphisms by PubMed identifiers</p>
<pre><code class="language-Python">from pandasgwas.get_variants import get_variants_by_pubmed_id
snps = get_variants_by_pubmed_id('21041247')
</code></pre>



<table class="field-list">
  <colgroup>
    <col class="field-name" />
    <col class="field-body" />
  </colgroup>
  <tbody valign="top">
    <tr class="field">
      <th class="field-name">Parameters:</th>
      <td class="field-body">
        <ul class="first simple">
            <li>
              <b><code>pubmed_id</code></b>
                  (<code>str</code>)
              –
              <div class="doc-md-description">
                <p>PubMed identifier</p>
              </div>
            </li>
            <li>
              <b><code>interactive</code></b>
                  (<code>bool</code>, default:
                      <code>True</code>
)
              –
              <div class="doc-md-description">
                <p>Indicates whether to run in interactive mode, when the value is True and the query data is divided into many pages, the function will prompt whether to continue the download.</p>
              </div>
            </li>
        </ul>
      </td>
    </tr>
  </tbody>
</table>


<table class="field-list">
  <colgroup>
    <col class="field-name" />
    <col class="field-body" />
  </colgroup>
  <tbody valign="top">
    <tr class="field">
    <th class="field-name">Returns:</th>
    <td class="field-body">
      <ul class="first simple">
          <li>
                <code><a class="autorefs autorefs-internal" title="pandasgwas.Variant.Variant" href="../Variant/#pandasgwas.Variant.Variant">Variant</a></code>
            –
            <div class="doc-md-description">
              <p>A Variant object</p>
            </div>
          </li>
      </ul>
    </td>
    </tr>
  </tbody>
</table>
  </div>

</div>


<div class="doc doc-object doc-function">



<h3 id="pandasgwas.get_variants.get_variants_by_reported_trait" class="doc doc-heading">
          <span class="doc doc-object-name doc-function-name">get_variants_by_reported_trait</span>


</h3>
<div class="doc-signature highlight"><pre><span></span><code><span class="n">get_variants_by_reported_trait</span><span class="p">(</span>
    <span class="n">reported_trait</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span> <span class="n">interactive</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">True</span>
<span class="p">)</span> <span class="o">-&gt;</span> <span class="n">Variant</span>
</code></pre></div>

  <div class="doc doc-contents ">
  
      <p>Get GWAS Catalog Single Nucleotide Polymorphisms that match the reported traits, as reported by the original authors of the study.</p>
<pre><code class="language-Python">from pandasgwas.get_variants import get_variants_by_reported_trait
snps = get_variants_by_reported_trait(&quot;Dupuytren's disease&quot;)
</code></pre>



<table class="field-list">
  <colgroup>
    <col class="field-name" />
    <col class="field-body" />
  </colgroup>
  <tbody valign="top">
    <tr class="field">
      <th class="field-name">Parameters:</th>
      <td class="field-body">
        <ul class="first simple">
            <li>
              <b><code>reported_trait</code></b>
                  (<code>str</code>)
              –
              <div class="doc-md-description">
                <p>Trait are reported by the original authors of the study</p>
              </div>
            </li>
            <li>
              <b><code>interactive</code></b>
                  (<code>bool</code>, default:
                      <code>True</code>
)
              –
              <div class="doc-md-description">
                <p>Indicates whether to run in interactive mode, when the value is True and the query data is divided into many pages, the function will prompt whether to continue the download.</p>
              </div>
            </li>
        </ul>
      </td>
    </tr>
  </tbody>
</table>


<table class="field-list">
  <colgroup>
    <col class="field-name" />
    <col class="field-body" />
  </colgroup>
  <tbody valign="top">
    <tr class="field">
    <th class="field-name">Returns:</th>
    <td class="field-body">
      <ul class="first simple">
          <li>
                <code><a class="autorefs autorefs-internal" title="pandasgwas.Variant.Variant" href="../Variant/#pandasgwas.Variant.Variant">Variant</a></code>
            –
            <div class="doc-md-description">
              <p>A Variant object</p>
            </div>
          </li>
      </ul>
    </td>
    </tr>
  </tbody>
</table>
  </div>

</div>


<div class="doc doc-object doc-function">



<h3 id="pandasgwas.get_variants.get_variants_by_study_id" class="doc doc-heading">
          <span class="doc doc-object-name doc-function-name">get_variants_by_study_id</span>


</h3>
<div class="doc-signature highlight"><pre><span></span><code><span class="n">get_variants_by_study_id</span><span class="p">(</span>
    <span class="n">study_id</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span> <span class="n">interactive</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">True</span>
<span class="p">)</span> <span class="o">-&gt;</span> <span class="n">Variant</span>
</code></pre></div>

  <div class="doc doc-contents ">
  
      <p>Get GWAS Catalog Single Nucleotide Polymorphisms by Study identifier</p>
<pre><code class="language-Python">from pandasgwas.get_variants import get_variants_by_study_id
snps = get_variants_by_study_id('GCST000854')
</code></pre>



<table class="field-list">
  <colgroup>
    <col class="field-name" />
    <col class="field-body" />
  </colgroup>
  <tbody valign="top">
    <tr class="field">
      <th class="field-name">Parameters:</th>
      <td class="field-body">
        <ul class="first simple">
            <li>
              <b><code>study_id</code></b>
                  (<code>str</code>)
              –
              <div class="doc-md-description">
                <p>Study identifier, accessionId in Study</p>
              </div>
            </li>
            <li>
              <b><code>interactive</code></b>
                  (<code>bool</code>, default:
                      <code>True</code>
)
              –
              <div class="doc-md-description">
                <p>Indicates whether to run in interactive mode, when the value is True and the query data is divided into many pages, the function will prompt whether to continue the download.</p>
              </div>
            </li>
        </ul>
      </td>
    </tr>
  </tbody>
</table>


<table class="field-list">
  <colgroup>
    <col class="field-name" />
    <col class="field-body" />
  </colgroup>
  <tbody valign="top">
    <tr class="field">
    <th class="field-name">Returns:</th>
    <td class="field-body">
      <ul class="first simple">
          <li>
                <code><a class="autorefs autorefs-internal" title="pandasgwas.Variant.Variant" href="../Variant/#pandasgwas.Variant.Variant">Variant</a></code>
            –
            <div class="doc-md-description">
              <p>A Variant object</p>
            </div>
          </li>
      </ul>
    </td>
    </tr>
  </tbody>
</table>
  </div>

</div>


<div class="doc doc-object doc-function">



<h3 id="pandasgwas.get_variants.get_variants_by_variant_id" class="doc doc-heading">
          <span class="doc doc-object-name doc-function-name">get_variants_by_variant_id</span>


</h3>
<div class="doc-signature highlight"><pre><span></span><code><span class="n">get_variants_by_variant_id</span><span class="p">(</span>
    <span class="n">variant_id</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span> <span class="n">interactive</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">True</span>
<span class="p">)</span> <span class="o">-&gt;</span> <span class="n">Variant</span>
</code></pre></div>

  <div class="doc doc-contents ">
  
      <p>Get GWAS Catalog Single Nucleotide Polymorphisms by Single Nucleotide Polymorphism identifier</p>
<pre><code class="language-Python">from pandasgwas.get_variants import get_variants_by_variant_id
snps = get_variants_by_variant_id('rs7744020')
</code></pre>



<table class="field-list">
  <colgroup>
    <col class="field-name" />
    <col class="field-body" />
  </colgroup>
  <tbody valign="top">
    <tr class="field">
      <th class="field-name">Parameters:</th>
      <td class="field-body">
        <ul class="first simple">
            <li>
              <b><code>variant_id</code></b>
                  (<code>str</code>)
              –
              <div class="doc-md-description">
                <p>Single Nucleotide Polymorphism identifier, rsId in Single Nucleotide Polymorphism</p>
              </div>
            </li>
            <li>
              <b><code>interactive</code></b>
                  (<code>bool</code>, default:
                      <code>True</code>
)
              –
              <div class="doc-md-description">
                <p>Indicates whether to run in interactive mode, when the value is True and the query data is divided into many pages, the function will prompt whether to continue the download.</p>
              </div>
            </li>
        </ul>
      </td>
    </tr>
  </tbody>
</table>


<table class="field-list">
  <colgroup>
    <col class="field-name" />
    <col class="field-body" />
  </colgroup>
  <tbody valign="top">
    <tr class="field">
    <th class="field-name">Returns:</th>
    <td class="field-body">
      <ul class="first simple">
          <li>
                <code><a class="autorefs autorefs-internal" title="pandasgwas.Variant.Variant" href="../Variant/#pandasgwas.Variant.Variant">Variant</a></code>
            –
            <div class="doc-md-description">
              <p>A Variant object</p>
            </div>
          </li>
      </ul>
    </td>
    </tr>
  </tbody>
</table>
  </div>

</div>



  </div>

  </div>

</div>
              
            </div>
          </div><footer>
    <div class="rst-footer-buttons" role="navigation" aria-label="Footer Navigation">
        <a href="../get_traits/" class="btn btn-neutral float-left" title="Get traits"><span class="icon icon-circle-arrow-left"></span> Previous</a>
        <a href="../set_operation/" class="btn btn-neutral float-right" title="Set operation">Next <span class="icon icon-circle-arrow-right"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
  </div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
          
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" aria-label="Versions">
  <span class="rst-current-version" data-toggle="rst-current-version">
    
    
      <span><a href="../get_traits/" style="color: #fcfcfc">&laquo; Previous</a></span>
    
    
      <span><a href="../set_operation/" style="color: #fcfcfc">Next &raquo;</a></span>
    
  </span>
</div>
    <script src="../js/jquery-3.6.0.min.js"></script>
    <script>var base_url = "..";</script>
    <script src="../js/theme_extra.js"></script>
    <script src="../js/theme.js"></script>
      <script src="../search/main.js"></script>
    <script>
        jQuery(function () {
            SphinxRtdTheme.Navigation.enable(true);
        });
    </script>

</body>
</html>
