<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
      <link rel="shortcut icon" href="../img/favicon.ico" />
    <title>get_variants - pandasGWAS Documentation</title>
    <link rel="stylesheet" href="../css/theme.css" />
    <link rel="stylesheet" href="../css/theme_extra.css" />
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.5.0/styles/github.min.css" />
    
      <script>
        // Current page data
        var mkdocs_page_name = "get_variants";
        var mkdocs_page_input_path = "get_variants.md";
        var mkdocs_page_url = null;
      </script>
    
    <script src="../js/jquery-3.6.0.min.js" defer></script>
    <!--[if lt IE 9]>
      <script src="../js/html5shiv.min.js"></script>
    <![endif]-->
      <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.5.0/highlight.min.js"></script>
      <script>hljs.initHighlightingOnLoad();</script> 
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
    <div class="wy-side-scroll">
      <div class="wy-side-nav-search">
          <a href=".." class="icon icon-home"> pandasGWAS Documentation
        </a><div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
      <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="..">pandasGWAS Documentation</a>
                </li>
              </ul>
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../Association/">Association</a>
                </li>
              </ul>
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../Browser/">Browser</a>
                </li>
              </ul>
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../Study/">Study</a>
                </li>
              </ul>
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../Trait/">Trait</a>
                </li>
              </ul>
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../Variant/">Variant</a>
                </li>
              </ul>
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../get_associations/">get_associations</a>
                </li>
              </ul>
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../get_studies/">get_studies</a>
                </li>
              </ul>
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../get_traits/">get_traits</a>
                </li>
              </ul>
              <ul class="current">
                <li class="toctree-l1 current"><a class="reference internal current" href="./">get_variants</a>
    <ul class="current">
    <li class="toctree-l2"><a class="reference internal" href="#function-get_variants_by_study_idstudy_id-str-interactive-bool-true-variant">function get_variants_by_study_id(study_id: str, interactive: bool = True) -&gt; Variant</a>
        <ul>
    <li class="toctree-l3"><a class="reference internal" href="#param-study_id">param study_id</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#param-interactive">param interactive</a>
    </li>
        </ul>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#function-get_variants_by_association_idassociation_id-str-interactive-bool-true-variant">function get_variants_by_association_id(association_id: str, interactive: bool = True) -&gt; Variant</a>
        <ul>
    <li class="toctree-l3"><a class="reference internal" href="#param-association_id">param association_id</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#param-interactive_1">param interactive</a>
    </li>
        </ul>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#function-get_variants_by_variant_idvariant_id-str-interactive-bool-true-variant">function get_variants_by_variant_id(variant_id: str, interactive: bool = True) -&gt; Variant</a>
        <ul>
    <li class="toctree-l3"><a class="reference internal" href="#param-variant_id">param variant_id</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#param-interactive_2">param interactive</a>
    </li>
        </ul>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#function-get_variants_by_efo_idefo_id-str-interactive-bool-true-variant">function get_variants_by_efo_id(efo_id: str, interactive: bool = True) -&gt; Variant</a>
        <ul>
    <li class="toctree-l3"><a class="reference internal" href="#param-efo_id">param efo_id</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#param-interactive_3">param interactive</a>
    </li>
        </ul>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#function-get_variants_by_pubmed_idpubmed_id-str-interactive-bool-true-variant">function get_variants_by_pubmed_id(pubmed_id: str, interactive: bool = True) -&gt; Variant</a>
        <ul>
    <li class="toctree-l3"><a class="reference internal" href="#param-pubmed_id">param pubmed_id</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#param-interactive_4">param interactive</a>
    </li>
        </ul>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#function-get_variants_by_genomic_rangechromosome-str-start-int-end-int-interactive-bool-true-variant">function get_variants_by_genomic_range(chromosome: str, start: int, end: int, interactive: bool = True) -&gt; Variant</a>
        <ul>
    <li class="toctree-l3"><a class="reference internal" href="#param-chromosome">param chromosome</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#param-start">param start</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#param-end">param end</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#param-interactive_5">param interactive</a>
    </li>
        </ul>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#function-get_variants_by_cytogenetic_bandcytogenetic_band-str-interactive-bool-true-variant">function get_variants_by_cytogenetic_band(cytogenetic_band: str, interactive: bool = True) -&gt; Variant</a>
        <ul>
    <li class="toctree-l3"><a class="reference internal" href="#param-cytogenetic_band">param cytogenetic_band</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#param-interactive_6">param interactive</a>
    </li>
        </ul>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#function-get_variants_by_gene_namegene_name-str-interactive-bool-true-variant">function get_variants_by_gene_name(gene_name: str, interactive: bool = True) -&gt; Variant</a>
        <ul>
    <li class="toctree-l3"><a class="reference internal" href="#param-gene_name">param gene_name</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#param-interactive_7">param interactive</a>
    </li>
        </ul>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#function-get_variants_by_efo_traitefo_trait-str-interactive-bool-true-variant">function get_variants_by_efo_trait(efo_trait: str, interactive: bool = True) -&gt; Variant</a>
        <ul>
    <li class="toctree-l3"><a class="reference internal" href="#param-efo_trait">param efo_trait</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#param-interactive_8">param interactive</a>
    </li>
        </ul>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#function-get_variants_by_reported_traitreported_trait-str-interactive-bool-true-variant">function get_variants_by_reported_trait(reported_trait: str, interactive: bool = True) -&gt; Variant</a>
        <ul>
    <li class="toctree-l3"><a class="reference internal" href="#param-reported_trait">param reported_trait</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#param-interactive_9">param interactive</a>
    </li>
        </ul>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#function-get_variants_allinteractive-bool-true-variant">function get_variants_all(interactive: bool = True) -&gt; Variant</a>
        <ul>
    <li class="toctree-l3"><a class="reference internal" href="#param-interactive_10">param interactive</a>
    </li>
        </ul>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#function-get_variantsstudy_id-str-none-association_id-str-none-variant_id-str-none-efo_id-str-none-pubmed_id-str-none-genomic_range-listint-none-gene_name-str-none-efo_trait-str-none-reported_trait-str-none-set_operation-str-bind-interactive-bool-true-variant">function get_variants(study_id: str = None, association_id: str = None, variant_id: str = None, efo_id: str = None, pubmed_id: str = None, genomic_range: List[int] = None, gene_name: str = None, efo_trait: str = None, reported_trait: str = None, set_operation: str = 'bind', interactive: bool = True) -&gt; Variant</a>
    </li>
    </ul>
                </li>
              </ul>
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../set_operation/">set_operation</a>
                </li>
              </ul>
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../summary_statistics/">summary_statistics</a>
                </li>
              </ul>
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../utility/">utility</a>
                </li>
              </ul>
      </div>
    </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">
      <nav class="wy-nav-top" role="navigation" aria-label="Mobile navigation menu">
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="..">pandasGWAS Documentation</a>
        
      </nav>
      <div class="wy-nav-content">
        <div class="rst-content"><div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href=".." class="icon icon-home" alt="Docs"></a> &raquo;</li>
      <li>get_variants</li>
    <li class="wy-breadcrumbs-aside">
    </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
            <div class="section" itemprop="articleBody">
              
                <h1 id="get_variants">get_variants</h1>
<p>Get Single Nucleotide Polymorphism function based on different search criteria</p>
<h2 id="function-get_variants_by_study_idstudy_id-str-interactive-bool-true-variant"><em>function</em> get_variants_by_study_id(study_id: str, interactive: bool = True) -&gt; Variant</h2>
<p>Get GWAS Catalog Single Nucleotide Polymorphisms by Study identifier</p>
<pre><code class="language-Python">from pandasgwas.get_variants import get_variants_by_study_id

snps = get_variants_by_study_id('GCST000854')
</code></pre>
<h3 id="param-study_id">param study_id</h3>
<p>Study identifier, accessionId in Study</p>
<h3 id="param-interactive">param interactive</h3>
<p>Indicates whether to run in interactive mode, when the value is True and the query data is divided into many pages, the function will prompt whether to continue the download.</p>
<h2 id="function-get_variants_by_association_idassociation_id-str-interactive-bool-true-variant"><em>function</em> get_variants_by_association_id(association_id: str, interactive: bool = True) -&gt; Variant</h2>
<p>Get GWAS Catalog Single Nucleotide Polymorphisms by by their Association identifier</p>
<pre><code class="language-Python">from pandasgwas.get_variants import get_variants_by_association_id

snps = get_variants_by_association_id('16603')
</code></pre>
<h3 id="param-association_id">param association_id</h3>
<p>Association identifier, associationId in Association</p>
<h3 id="param-interactive_1">param interactive</h3>
<p>Indicates whether to run in interactive mode, when the value is True and the query data is divided into many pages, the function will prompt whether to continue the download.</p>
<h2 id="function-get_variants_by_variant_idvariant_id-str-interactive-bool-true-variant"><em>function</em> get_variants_by_variant_id(variant_id: str, interactive: bool = True) -&gt; Variant</h2>
<p>Get GWAS Catalog Single Nucleotide Polymorphisms by Single Nucleotide Polymorphism identifier</p>
<pre><code class="language-Python">from pandasgwas.get_variants import get_variants_by_variant_id

snps = get_variants_by_variant_id('rs7744020')
</code></pre>
<h3 id="param-variant_id">param variant_id</h3>
<p>Single Nucleotide Polymorphism identifier, rsId in Single Nucleotide Polymorphism</p>
<h3 id="param-interactive_2">param interactive</h3>
<p>Indicates whether to run in interactive mode, when the value is True and the query data is divided into many pages, the function will prompt whether to continue the download.</p>
<h2 id="function-get_variants_by_efo_idefo_id-str-interactive-bool-true-variant"><em>function</em> get_variants_by_efo_id(efo_id: str, interactive: bool = True) -&gt; Variant</h2>
<p>Get GWAS Catalog Single Nucleotide Polymorphisms by their EFO Trait identifier</p>
<pre><code class="language-Python">from pandasgwas.get_variants import get_variants_by_efo_id

snps = get_variants_by_efo_id('EFO_0001065')
</code></pre>
<h3 id="param-efo_id">param efo_id</h3>
<p>EFO Trait identifier, shortForm in EFO Trait</p>
<h3 id="param-interactive_3">param interactive</h3>
<p>Indicates whether to run in interactive mode, when the value is True and the query data is divided into many pages, the function will prompt whether to continue the download.</p>
<h2 id="function-get_variants_by_pubmed_idpubmed_id-str-interactive-bool-true-variant"><em>function</em> get_variants_by_pubmed_id(pubmed_id: str, interactive: bool = True) -&gt; Variant</h2>
<p>Get GWAS Catalog Single Nucleotide Polymorphisms by PubMed identifiers</p>
<pre><code class="language-Python">from pandasgwas.get_variants import get_variants_by_pubmed_id

snps = get_variants_by_pubmed_id('21041247')
</code></pre>
<h3 id="param-pubmed_id">param pubmed_id</h3>
<p>PubMed identifier</p>
<h3 id="param-interactive_4">param interactive</h3>
<p>Indicates whether to run in interactive mode, when the value is True and the query data is divided into many pages, the function will prompt whether to continue the download.</p>
<h2 id="function-get_variants_by_genomic_rangechromosome-str-start-int-end-int-interactive-bool-true-variant"><em>function</em> get_variants_by_genomic_range(chromosome: str, start: int, end: int, interactive: bool = True) -&gt; Variant</h2>
<p>Get GWAS Catalog Single Nucleotide Polymorphisms</p>
<pre><code class="language-Python">from pandasgwas.get_variants import get_variants_by_genomic_range

snps = get_variants_by_genomic_range('1', 2300001, 5300000)
</code></pre>
<h3 id="param-chromosome">param chromosome</h3>
<p>Human chromosome names: autosomal and sexual chromosomes only, i.e., 1--22, X and Y</p>
<h3 id="param-start">param start</h3>
<p>Start position of range (starts at 1).</p>
<h3 id="param-end">param end</h3>
<p>End position of range (inclusive).</p>
<h3 id="param-interactive_5">param interactive</h3>
<p>Indicates whether to run in interactive mode, when the value is True and the query data is divided into many pages, the function will prompt whether to continue the download.</p>
<h2 id="function-get_variants_by_cytogenetic_bandcytogenetic_band-str-interactive-bool-true-variant"><em>function</em> get_variants_by_cytogenetic_band(cytogenetic_band: str, interactive: bool = True) -&gt; Variant</h2>
<p>Get GWAS Catalog Single Nucleotide Polymorphisms</p>
<pre><code class="language-Python">from pandasgwas.get_variants import get_variants_by_cytogenetic_band

snps = get_variants_by_cytogenetic_band('1p36.32')
</code></pre>
<h3 id="param-cytogenetic_band">param cytogenetic_band</h3>
<p>Cytogenetic bands of the form '1p36.11'</p>
<h3 id="param-interactive_6">param interactive</h3>
<p>Indicates whether to run in interactive mode, when the value is True and the query data is divided into many pages, the function will prompt whether to continue the download.</p>
<h2 id="function-get_variants_by_gene_namegene_name-str-interactive-bool-true-variant"><em>function</em> get_variants_by_gene_name(gene_name: str, interactive: bool = True) -&gt; Variant</h2>
<p>Get GWAS Catalog Single Nucleotide Polymorphisms</p>
<pre><code class="language-Python">from pandasgwas.get_variants import get_variants_by_gene_name

snps = get_variants_by_gene_name('KIAA0319')
</code></pre>
<h3 id="param-gene_name">param gene_name</h3>
<p>Gene names</p>
<h3 id="param-interactive_7">param interactive</h3>
<p>Indicates whether to run in interactive mode, when the value is True and the query data is divided into many pages, the function will prompt whether to continue the download.</p>
<h2 id="function-get_variants_by_efo_traitefo_trait-str-interactive-bool-true-variant"><em>function</em> get_variants_by_efo_trait(efo_trait: str, interactive: bool = True) -&gt; Variant</h2>
<p>Get GWAS Catalog Single Nucleotide Polymorphisms that match trait description</p>
<pre><code class="language-Python">from pandasgwas.get_variants import get_variants_by_efo_trait

snps = get_variants_by_efo_trait('MHPG measurement')
</code></pre>
<h3 id="param-efo_trait">param efo_trait</h3>
<p>Trait description</p>
<h3 id="param-interactive_8">param interactive</h3>
<p>Indicates whether to run in interactive mode, when the value is True and the query data is divided into many pages, the function will prompt whether to continue the download.</p>
<h2 id="function-get_variants_by_reported_traitreported_trait-str-interactive-bool-true-variant"><em>function</em> get_variants_by_reported_trait(reported_trait: str, interactive: bool = True) -&gt; Variant</h2>
<p>Get GWAS Catalog Single Nucleotide Polymorphisms that match the reported traits, as reported by the original authors of the study.</p>
<pre><code class="language-Python">from pandasgwas.get_variants import get_variants_by_reported_trait

snps = get_variants_by_reported_trait(&quot;Dupuytren's disease&quot;)
</code></pre>
<h3 id="param-reported_trait">param reported_trait</h3>
<p>Trait are reported by the original authors of the study</p>
<h3 id="param-interactive_9">param interactive</h3>
<p>Indicates whether to run in interactive mode, when the value is True and the query data is divided into many pages, the function will prompt whether to continue the download.</p>
<h2 id="function-get_variants_allinteractive-bool-true-variant"><em>function</em> get_variants_all(interactive: bool = True) -&gt; Variant</h2>
<p>Get all Single Nucleotide Polymorphisms</p>
<pre><code class="language-Python">from pandasgwas.get_variants import get_variants_all

snps = get_variants_all()
</code></pre>
<h3 id="param-interactive_10">param interactive</h3>
<p>Indicates whether to run in interactive mode, when the value is True and the query data is divided into many pages, the function will prompt whether to continue the download.</p>
<h2 id="function-get_variantsstudy_id-str-none-association_id-str-none-variant_id-str-none-efo_id-str-none-pubmed_id-str-none-genomic_range-listint-none-gene_name-str-none-efo_trait-str-none-reported_trait-str-none-set_operation-str-bind-interactive-bool-true-variant"><em>function</em> get_variants(study_id: str = None, association_id: str = None, variant_id: str = None, efo_id: str = None, pubmed_id: str = None, genomic_range: List[int] = None, gene_name: str = None, efo_trait: str = None, reported_trait: str = None, set_operation: str = 'bind', interactive: bool = True) -&gt; Variant</h2>
<p>Retrieves Single Nucleotide Polymorphisms via the NHGRI-EBI GWAS Catalog REST API. The REST API is queried multiple times with the criteria passed as arguments. By default all Single Nucleotide Polymorphisms that match the criteria supplied in the arguments are retrieved: this corresponds to the default set_operation set to 'bind', If you rather have only the Single Nucleotide Polymorphisms that match simultaneously all criteria provided, then set set_operation to 'intersection'.</p>
              
            </div>
          </div><footer>
    <div class="rst-footer-buttons" role="navigation" aria-label="Footer Navigation">
        <a href="../get_traits/" class="btn btn-neutral float-left" title="get_traits"><span class="icon icon-circle-arrow-left"></span> Previous</a>
        <a href="../set_operation/" class="btn btn-neutral float-right" title="set_operation">Next <span class="icon icon-circle-arrow-right"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
  </div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
          
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" aria-label="Versions">
  <span class="rst-current-version" data-toggle="rst-current-version">
    
    
      <span><a href="../get_traits/" style="color: #fcfcfc">&laquo; Previous</a></span>
    
    
      <span><a href="../set_operation/" style="color: #fcfcfc">Next &raquo;</a></span>
    
  </span>
</div>
    <script>var base_url = '..';</script>
    <script src="../js/theme_extra.js" defer></script>
    <script src="../js/theme.js" defer></script>
      <script src="../search/main.js" defer></script>
    <script defer>
        window.onload = function () {
            SphinxRtdTheme.Navigation.enable(true);
        };
    </script>

</body>
</html>
